internal/worker/worker.go

```diff
@@ -198,6 +199,10 @@ func (s *Worker) Run() error {
 
 	go s.manageWorkerCapacity()
 	go s.processStopContainerEvents()
+	defer func() {
+		close(s.completedRequests)
```

Comment:
Would these be better added in their respective goroutines?

Reply:
How would that look?

Reply:
Nevermind, I took another look at these 2 functions. This looks good